File: frontend/lib/medical_record.ts
import api from "./api";

export interface MedicalRecord {
  id?: number;

  patient_id: number;
  doctor_id: number;
  appointment_id: number;

  symptoms: string;
  diagnosis: string;

  medications?: string;
  allergies?: string;
  doctor_notes?: string;

  created_at?: string;
  updated_at?: string;
}


export async function getMedicalRecords(
  token: string,
  search = ""
) {
  const response = await api.get(
    "/medical-records",
    {
      params: {
        search,
      },
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}


export async function createMedicalRecord(
  record: MedicalRecord,
  token: string
) {
  const response = await api.post(
    "/medical-records",
    record,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}


export async function updateMedicalRecord(
  id: number,
  record: MedicalRecord,
  token: string
) {
  const response = await api.put(
    `/medical-records/${id}`,
    record,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );

  return response.data;
}


export async function deleteMedicalRecord(
  id: number,
  token: string
) {
  await api.delete(
    `/medical-records/${id}`,
    {
      headers: {
        Authorization: `Bearer ${token}`,
      },
    }
  );
}

File: backend/app/schemas/medical_record.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class MedicalRecordCreate(BaseModel):
    patient_id: int
    doctor_id: int
    appointment_id: int

    symptoms: str

    diagnosis: str

    medications: Optional[str] = None

    allergies: Optional[str] = None

    doctor_notes: Optional[str] = None


class MedicalRecordUpdate(BaseModel):
    symptoms: Optional[str] = None

    diagnosis: Optional[str] = None

    medications: Optional[str] = None

    allergies: Optional[str] = None

    doctor_notes: Optional[str] = None


class MedicalRecordRead(BaseModel):
    id: int

    patient_id: int

    doctor_id: int

    appointment_id: int

    symptoms: str

    diagnosis: str

    medications: Optional[str]

    allergies: Optional[str]

    doctor_notes: Optional[str]

    created_at: datetime

    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True
    )